### PR TITLE
CA-187179: Refine error message of pool auth disable

### PR DESCRIPTION
**Pull request by cheng-z**
When disable external auth, although failed, the external auth been disabled.
So refine the error message.

Signed-off-by: Cheng Zhang <cheng.zhang@citrix.com>

### CODE REVIEW

**Comment by cheng-z** [comment]
Hi @robhoes 
Could you please help to review the code?
Thanks.

File: ocaml/idl/datamodel.ml [hunk 1]
```diff
@@ -928,9 +928,11 @@ let _ =
   error Api_errors.pool_auth_disable_failed ["host";"message"]
     ~doc:"The pool failed to disable the external authentication of at least one host." ();
   error Api_errors.pool_auth_disable_failed_wrong_credentials ["host";"message"]
-    ~doc:"The pool failed to disable the external authentication of at least one host." ();
+    ~doc:"External authentication has been disabled with errors." ();
   error Api_errors.pool_auth_disable_failed_permission_denied ["host";"message"]
-    ~doc:"The pool failed to disable the external authentication of at least one host." ();
+    ~doc:"External authentication has been disabled with errors." ();
+  error Api_errors.pool_auth_disable_failed_invalid_account ["host";"message"]
+      ~doc:"External authentication has been disabled with errors." ();
```
Comment (robhoes) [inline ocaml/idl/datamodel.ml]: These three all have the same doc string, but the error code has more info that could be used in the text.

Also the indentation of this last line is slightly off.

File: ocaml/xapi-consts/api_errors.ml [hunk 2]
```diff
@@ -493,6 +493,7 @@ let pool_auth_enable_failed_duplicate_hostname = pool_auth_enable_failed^"_DUPLI
 let pool_auth_disable_failed = pool_auth_prefix^auth_disable_failed
 let pool_auth_disable_failed_wrong_credentials = pool_auth_disable_failed^auth_suffix_wrong_credentials
 let pool_auth_disable_failed_permission_denied = pool_auth_disable_failed^auth_suffix_permission_denied
+let pool_auth_disable_failed_invalid_account = pool_auth_disable_failed^auth_suffix_invalid_account
```
Comment (robhoes) [inline ocaml/xapi-consts/api_errors.ml]: Where is this new error raised?

Comment (cheng-z) [inline ocaml/xapi-consts/api_errors.ml]: The error actually can be found also in https://github.com/xapi-project/xen-api/commit/7abecc42eb44ed3e00fdf1648c9d588e0a86d618

**Comment by cheng-z** [comment]
Hi @robhoes 
I've update the error message to use the same description as XenCenter does. Please help to have a look again.
Also the indentation issue been fixed.

Please help to review the code again and merge if no other concern.
Thanks.

**Pull request merged**